src/core/components/model.jsx

```diff
@@ -224,6 +224,7 @@ class Model extends Component {
 
     if ( schema && (schema.get("type") || schema.get("properties")) ) {
       modelSchema = schema
+      modelName = name
```

Comment:
Thanks for opening this PR!

I'd prefer to see the name prop value defined as `name || modelName`, rather than reassign the value of `modelName`. I'm fairly certain doing so is effectively equivalent behavior, and it feels less magical than putting a reassignment in an `if`.

Let me know what you think, @GSokol.